pkg/kubectl/cmd/util/factory.go

```diff
@@ -346,6 +351,15 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		},
 		Describer: func(mapping *meta.RESTMapping) (kubectl.Describer, error) {
 			mappingVersion := mapping.GroupVersionKind.GroupVersion()
+			if mapping.GroupVersionKind.Group == federation.GroupName {
```

Comment:
To clarify, I dont want you to change that in this PR. Consider the above as a thought that we can discuss more.